extensions/modules/file/src/main/java/org/exist/xquery/modules/file/Sync.java

```diff
@@ -55,183 +62,356 @@
-			output.addAttribute(new QName("name", XMLConstants.NULL_NS_URI), doc.getFileURI().toString());
-			output.addAttribute(new QName("collection", XMLConstants.NULL_NS_URI), doc.getCollection().getURI().toString());
-			output.addAttribute(new QName("type", XMLConstants.NULL_NS_URI), "xml");
-			output.addAttribute(new QName("modified", XMLConstants.NULL_NS_URI), new DateTimeValue(new Date(doc.getLastModified())).getStringValue());
+    }
+
+    public Sync(final XQueryContext context, final FunctionSignature signature) {
+        super(context, signature);
+    }
+
+    @Override
+    public Sequence eval(final Sequence[] args, final Sequence contextSequence) throws XPathException {
+
+        if (!context.getSubject().hasDbaRole()) {
+            throw new XPathException(this, "Function file:sync is only available to the DBA role");
+        }
+
+        final String collectionPath = args[0].getStringValue();
+        final String target = args[1].getStringValue();
+        final HashMap<String, Sequence> options = getOptions(args[2]);
+
+        return startSync(target, collectionPath, options);
+    }
+
+    private HashMap<String, Sequence> getOptions(final Sequence parameter) throws XPathException {
+        HashMap<String, Sequence> options = new HashMap<>();
+        options.put(AFTER_OPT, Sequence.EMPTY_SEQUENCE);
+        options.put(PRUNE_OPT, new BooleanValue(false));
+        options.put(EXCLUDES_OPT, Sequence.EMPTY_SEQUENCE);
+
+        if (parameter.isEmpty()) {
+            return options;
+        }
+
+        final Item item = parameter.itemAt(0);
+
+        if (item.getType() == Type.MAP) {
+            final AbstractMapType optionsMap = (AbstractMapType) item;
+
+            final Sequence seq = optionsMap.get(new StringValue(EXCLUDES_OPT));
+            if (!seq.isEmpty() && seq.getItemType() != Type.STRING) {
+                throw new XPathException(this, ErrorCodes.XPTY0004,
+                        "Invalid value for option \"excludes\", expected xs:string* got " +
+                                Type.getTypeName(seq.getItemType()));
+            }
+            options.put(EXCLUDES_OPT, seq);
+
+            checkOption(optionsMap, PRUNE_OPT, Type.BOOLEAN, options);
+            checkOption(optionsMap, AFTER_OPT, Type.DATE_TIME, options);
+        } else if (parameter.itemAt(0).getType() == Type.DATE_TIME) {
+            // QUESTION(JL) LOG deprecation warning?
+            options.put(AFTER_OPT, parameter);
+        } else {
+            throw new XPathException(this, ErrorCodes.XPTY0004,
+                    "Invalid 3rd parameter, allowed are xs:dateTime or map(*) got " + Type.getTypeName(item.getType()));
+        }
+        return options;
+    }
+
+    private void checkOption(
+            final AbstractMapType optionsMap,
+            final String name,
+            final int expectedType,
+            final HashMap<String, Sequence> options
```

Comment:
For the parameter, use the lowest-common Interface, i.e. `Map` and not `HashMap`.